bin/programs/client/src/l2/executor/mod.rs

```diff
@@ -195,28 +195,47 @@ where
 
             // Create receipt envelope.
             let logs_bloom = logs_bloom(result.logs());
-            let receipt_envelope = wrap_receipt_with_bloom(
-                OpReceiptWithBloom {
-                    receipt: OpReceipt {
-                        status: result.is_success(),
-                        cumulative_gas_used: cumulative_gas_used as u128,
-                        logs: result.into_logs(),
-                        deposit_nonce: depositor
-                            .as_ref()
-                            .map(|depositor| depositor.account_info().unwrap_or_default().nonce),
-                        // The deposit receipt version was introduced in Canyon to indicate an
-                        // update to how receipt hashes should be computed
-                        // when set. The state transition process
-                        // ensures this is only set for post-Canyon deposit transactions.
-                        deposit_receipt_version: depositor
-                            .is_some()
-                            .then(|| self.config.is_canyon_active(payload.timestamp).then_some(1))
-                            .flatten(),
-                    },
+            let inner_receipt = Receipt {
+                status: Eip658Value::Eip658(result.is_success()),
+                cumulative_gas_used: cumulative_gas_used as u128,
+                logs: result.into_logs(),
+            };
+            let receipt_envelope = match transaction.tx_type() {
+                OpTxType::Legacy => OpReceiptEnvelope::Legacy(ReceiptWithBloom {
```

Comment:
hmm - pedantic nit, but can we keep this broken out like it was before (wrap_receipt_with_bloom)? Lots of inlining in this codeblock.